### ISSUE TITLE
Blender 3.3 Linux std:bad_cast

### ISSUE THREAD

**Issue by oculardegen** [LuxCoreRender/BlendLuxCore#787]
Steps to reproduce:
1. Enable LuxCore
2. Under Devices -> GPU Devices click "Update Device List"

Screenshots or rendered images:

Error message from Blender console:
RuntimeError: std::bad_cast

OS: Linux Ubuntu
Blender version: Blender 3.3
LuxCore version: Latest Developmental Version

**Comment by XanaDublaKublaConch** [comment]
![Screenshot_20220917_004646](https://user-images.githubusercontent.com/48071895/190840898-b92ec1f0-d381-4738-a539-7d9991f3f4ae.png)

**Comment by odil24** [comment]
@oculardegen @XanaDublaKublaConch are you have tried to delete startup config blend file from the main data folders? If no, so please try to delete and then again run fresh Blender with factory settings. Then try to enable again.

**Comment by XanaDublaKublaConch** [comment]
> @oculardegen @XanaDublaKublaConch are you have tried to delete startup config blend file from the main data folders? If no, so please try to delete and then again run fresh Blender with factory settings. Then try to enable again.

Yes. I moved my whole ~/.config/blender/ folder to old-blender/ and restarted Blender. 

- started Blender with factory settings
- ran through first time wizard in Blender
- installed the alpha release of BlendLuxCore
- enabled the addon
- Checked OpenCL/CUDA settings in the addons, but only CPU is available
- went to render settings
- selected LuxCore
- Selected GPU
- Expanded devices
- Click Update Device List
- got the same python error
Just to be sure Blender sees the GPU:
- Checked Edit->Preferences->system->Cycles Render Settings
- Cycles sees the NVidia RTX 3050 laptop GPU.

**Comment by XanaDublaKublaConch** [comment]
That didn't work either, but downgrading my nvidia driver from 515 back to 500 worked.

**Comment by odil24** [comment]
If you still have an issues maybe are you need to try this build?
https://github.com/LuxCoreRender/BlendLuxCore/releases/tag/blendluxcore_v2.6alpha1

**Comment by CodeFHD** [comment]
Closing this issue after a review with the following reasons/conclusions:
- Issue affected old Blender + LuxCore versions that are no longer supported
- Issue was likely driver issue based on fix reported by one user (i.e. can't be fixed in BlendLuxCore)

If you feel this issue is still relevant in 2025, please re-test it with a recent Blender + LuxCore version, then ask to reopen and provide the following information:
- Operating System
- CPU + GPU model
- Blender and LuxCoreRender version used for re-test
- Detailed steps how to reproduce the bug from an empty scene,
- provide a specific .blend-file
